[IMP] fiscal_localization/italy: add configuration steps for fiscal p… (#11485)
odoo/documentation: content/applications/finance/fiscal_localizations/italy.rst

the :abbr:`PA (Public Administration)` business directly, sending them both the invoice and the
received deadline notification by email. You can make an arrangement with them and manually set the
correct :guilabel:`SdI State` on your invoice.

Point of Sale in Italy: Fiscal printer
======================================

Starting with Odoo 18.0, we introduced the support for Italian fiscal printers for your point of
sale.
These printers often come with the default setting to use HTTP. In order for such a printer to work
with Odoo, that setting has to be changed to enable the printer to support HTTPS.

EPSON FP90-III
--------------

To setup this printer, you will need the configuration software EpsonFPWizard or the keyboard
connected to the printer. With the keyboard, you will then need to perform the following:

> #. Make sure you are on the default state of your fiscal printer
> #. Type **3333**
> #. Press **Chiave**
> #. Type **34**
> #. Press **Contante** until the screen displays **SSL**
> #. Press X to turn the value **0** into a **1**
> #. Press **Contante** 3 times to confirm
> #. Press **Chiave**

Reviewer: I made some suggestions in the following list:
- to avoid the unnecessary "you/your" 
- :guilabel: markup should be used to identify any text of the interactive user interface and literal to write technical terms or a specific value to insert.
- we usually try to start with the intent ("To do this, do yyy, and select zzz (for example)).

Also, as we tend to be as precise as possible in the instructions, could you add the name of the fields/windows where the information is to be typed or clicked whenever possible? Thank you!
```suggestion
#. Make sure that the fiscal printer is set to its default state.
#. Type `3333`.
#. Click :guilabel:`Chiave`.
#. Type `34`.
#. To display :guilabel:`SSL` on the screen, click :guilabel:`Contante` repeatedly.
#. To turn the value `0` into a `1`, click :guilabel:`X`.
#. To confirm, click :guilabel:`Contante` 3 times.
#. Click :guilabel:`Chiave`.
```

Reviewer: Hi @viso-odoo Thanks for the clarification. You're right, I think that it won't be possible to be more specific then. But, yes, I'll be in GR3 tomorrow morning, I'll contact you so you can show me! Thank you :)

> After these steps are completed, you need to log in to the printer with the device that runs the
> :ref:`point of sale <sales/point_of_sale>` so it recognises your printer's certificate.

Reviewer: ```suggestion
Then, log in to the printer with the device that runs the :ref:`point of sale <sales/point_of_sale>`
so it recognizes the printer's certificate.
```

> #. Navigate to your printer by typing `https://<ip-of-your-printer>` in a web browser search bar. You
> will be presented with a security prompt.
> #. Click **advanced**
> #. Click **proceed**

Reviewer: As we tend to be as precise as possible in the instructions, the intent for the following actions should be added to the top of the list.
Can you also add some details for steps 2 & 3 (the intent, the field or the window name)
```suggestion
#. To access the printer, open a web browser and enter `https://<ip-of-your-printer>` in the address
bar. A security message will then appear.
#. Click :guilabel:`advanced`
#. Click :guilabel:`proceed`
```

Author: @afma-odoo for this one i thought about inserting a picture of the webpage that appears (the one displaying the security message)
Would it be a valuable addition?
I took your comments into account for the rest

Reviewer: @viso-odoo An image shouldn't add information, as text instructions should be self-explanatory without relying on images. Images can, of course, be included to help the user better understand the content, but they tend to become outdated quickly ;) So, they should only be used sparingly and only when they provide clear added value. If you think it should be useful, you can add it!

> Once this is done, make sure your :ref:`point of sale <sales/point_of_sale>` is configured to work
> with the fiscal printer. Navigate to
> :menuselection:`Point of Sale --> Configuration --> Settings`.
> Scroll down to the section **Connected Devices**, type in the IP address of your printer and enable
> **Use HTTPS**.

Reviewer: ```suggestion
Then, to ensure the :ref:`point of sale <sales/point_of_sale>` is configured with the fiscal 
printer, go to :menuselection:`Point of Sale --> Configuration --> Settings`. In the 
:guilabel:`Connected Devices` section, add the IP address in the :guilabel:`Italian Fiscal Printer 
IP address` and enable :guilabel:`Use HTTPS`.
```

#. Press :guilabel:`Contante` 3 times until the screen displays :guilabel:`Web Server: SSL`.
#. To turn the value `0` into a `1`, press :guilabel:`X`.
#. To confirm, press :guilabel:`Contante` 3 times.
#. Press :guilabel:`Chiave`.

Then, log in to the printer with the device that runs the :doc:`point of sale
</applications/sales/point_of_sale>` so it recognizes the printer's certificate.

To approve and install the printer's certificate, follow these steps:

#. To access the printer, open a web browser and enter `https://<ip-of-your-printer>` in the address
   bar. A :guilabel:`Warning: Potential Security Risk Ahead` security message will then appear.
#. To show the certificate approval options, click :guilabel:`Advanced`.
#. To validate the certificate, click :guilabel:`Proceed`.

Then, to ensure the :doc:`point of sale </applications/sales/point_of_sale>` is configured with the
fiscal printer, go to :menuselection:`Point of Sale --> Configuration --> Settings`. In the
:guilabel:`Connected Devices` section, add the IP address in the :guilabel:`Italian Fiscal Printer
IP address` and enable :guilabel:`Use HTTPS`.

Configuring Simulation Mode

> ---------------------------

Reviewer: I think this part should come before the "Setting up the printer to work with Odoo" one to avoid users working in production mode without trying the simulation one.

It is possible to test the setup of the fiscal printer with Odoo. To do this, configure the fiscal
printer in Simulation mode with the following steps:

#. Make sure that the fiscal printer is set to its default state: printer on, start-up cycle
   complete, and no transaction in progress.
#. Type `3333`.
#. Press :guilabel:`Chiave`. The screen will display :guilabel:`Scleta Funzione`.
#. Type `14`. The screen will display :guilabel:`Apprendimento`.
#. Type `62`. The screen displays :guilabel:`Simulazione`.
#. To turn the :guilabel:`no` into a :guilabel:`si`, press :guilabel:`X`.
#. To confirm, press :guilabel:`Contante`.
#. Press :guilabel:`Chiave`.

To configure the printer to production, follow the above setps once more.

..note::
   In oder to test the printer configuration, the actual physical unit must be acquired and
   registered with the authorities.

> ..warning::
>    As simulation mode does send the data to the authorities, setting simulation should only be done
>    at the very beginning of the configuration of the printer. Once the printer has been sey to
>    production, it should not be set to simulation again.

Reviewer: As this is an important piece of information to give from the start, I'd prefer to add it immediately after the "Configuring Simulation Mode" heading.
```suggestion
.. warning::
   Since simulation mode transmits data to the authorities, it should only be enabled at the very 
   start of the printer's configuration process. After the printer is switched to production mode,
   it must not be reverted to simulation mode.
```
